packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,2896 @@
+  const MenuItemButton({
+    super.key,
+    this.shortcut,
+    this.onSelected,
+    this.onSelectedIntent,
+    this.onHover,
+    this.focusNode,
+    this.leadingIcon,
+    this.trailingIcon,
+    this.semanticsLabel,
+    this.backgroundColor,
+    this.foregroundColor,
+    this.overlayColor,
+    this.textStyle,
+    this.padding,
+    this.shape,
+    required this.label,
+  }) : assert(onSelected == null || onSelectedIntent == null,
+            'Only one of onSelected or onSelectedIntent may be specified');
+
+  /// A required widget displaying the label for this item in the menu.
+  final Widget label;
+
+  /// The optional shortcut that selects this [MenuItemButton].
+  ///
+  /// This shortcut is only enabled when [onSelected] is set.
+  final MenuSerializableShortcut? shortcut;
+
+  /// The function called when the mouse leaves or enters this menu item's
+  /// button.
+  final ValueChanged<bool>? onHover;
+
+  /// Returns a callback, if any, to be invoked if the platform menu receives a
+  /// "Menu.selectedCallback" method call from the platform for this item.
+  ///
+  /// Only items that do not have submenus will have this callback invoked.
+  ///
+  /// Only one of [onSelected] or [onSelectedIntent] may be specified.
+  ///
+  /// If neither [onSelected] nor [onSelectedIntent] are specified, then this
+  /// menu item is considered to be disabled.
+  ///
+  /// The default implementation returns null.
+  final VoidCallback? onSelected;
+
+  /// Returns an intent, if any, to be invoked if the platform receives a
+  /// "Menu.selectedCallback" method call from the platform for this item.
+  ///
+  /// Only items that do not have submenus will have this intent invoked.
+  ///
+  /// Only one of [onSelected] or [onSelectedIntent] may be specified.
+  ///
+  /// If neither [onSelected] nor [onSelectedIntent] are specified, then this
+  /// menu item is considered to be disabled.
+  ///
+  /// The default implementation returns null.
+  final Intent? onSelectedIntent;
+
+  /// The focus node to use for the menu item button.
+  final FocusNode? focusNode;
+
+  /// An optional icon to display before the [label] label.
+  final Widget? leadingIcon;
+
+  /// An optional icon to display after the [label] label.
+  final Widget? trailingIcon;
+
+  /// The semantic label of the menu item used by accessibility frameworks to
+  /// announce its label when the menu is focused.
+  ///
+  /// This semantics information will take precedence over semantics information
+  /// provided in [label].
+  final String? semanticsLabel;
+
+  /// The background color for this [MenuItemButton].
+  ///
+  /// Defaults to the ambient [Theme]'s [ColorScheme.surface] if null.
+  ///
+  /// See also:
+  ///
+  /// * [MenuThemeData.itemBackgroundColor], for the value in the [MenuTheme]
+  ///   that can be set instead of this property.
+  final MaterialStateProperty<Color?>? backgroundColor;
+
+  /// The foreground color for this [MenuItemButton].
```

Comment:
I suppose it goes without saying that this applies to icons? Is it also used to create a DefaultTextStyle?